### ISSUE TITLE
Exception when two successive string format placeholders and value of first one ends in `%`

### ISSUE THREAD

**Issue by Booksbaum** [fable-compiler/Fable#2398]
### Description

Exception when two successive string format placeholders and value of first one ends in `%`

### Repro code

```fsharp
printfn "%c%s" '%' "text"
```
([repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAY2QZ0fA5MnSUApgB5SJA&html=Q&css=Q))

### Expected and actual results

Expected Output: 
```
%text
```

Actual Output:
```
%%s
> Uncaught TypeError: toConsole(...)(...) is not a function
```

When the first formatter evaluates to something with `%` in last place, it seems to mess up an immediately following formatter.

This happens for all placeholders for `%`: 
* `%c`: `printfn "%c%s" '%' "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAY2QZ0fA5MnSUApgB5SJA&html=Q&css=Q)
* `%s`: `printfn "%s%s" "%" "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAZzYpzOKgUwA8pEg&html=Q&css=Q)
* `%A`: `printfn "%A%s" '%' "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAQWQZ0fA5MnSUApgB5SJA&html=Q&css=Q)
* width: `printfn "%5c%s" '%' "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAVgMbIM6PgcmTySgFMAPKRIA&html=Q&css=Q)

For all format placeholders following `%`:
* `%s`: `printfn "%c%s" '%' "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAY2QZ0fA5MnSUApgB5SJA&html=Q&css=Q)
* `%A`: `printfn "%c%A" '%' "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAY2QZ0fA5MnSUApgB5SJA&html=Q&css=Q)
* `%i`: `printfn "%c%i" '%' 42` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAY2QZ0fA5MnSUApgB5SJA&html=Q&css=Q)
* width: `printfn "%c%5i" '%' 42` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAY2QZ0fA5MnSUApgB5SJA&html=Q&css=Q)

For `%` in last place in value of first format placeholder:
* `printfn "%s%s" "%%" "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAZzYpzmcVApgB5SJA&html=Q&css=Q)
* `printfn "%s%s" "%s%" "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAZzYpbmcVApgB5SJA&html=Q&css=Q)
* `printfn "%s%s" "s%" "text"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAZzYp7OKgUwA8pEg&html=Q&css=Q)

It doesn't happen when:
* Space between the two formatters: `printfn "%c %s" '%' "test"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAY3sgzo+ByZuSUApgB5SJA&html=Q&css=Q)
* `%` isn't last char: `printfn "%s%s" "%c" "test"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAZzYpyDGnFQFMAPKRIA&html=Q&css=Q)
* `%` is left aligned (with big enough width to emit space(s) after `%`): `printfn "%-5c%s" '%' "test"` [repl](https://fable.io/repl/#?code=A4JwlgdgLgZhAEAiApAWgKwGNkGdHwHJkCkoBTADykSA&html=Q&css=Q)

### Related information

* Versions: 
  * Fable: `3.1.5`
  * REPL: `3.1.5`

**Comment by alfonsogarciacaro** [comment]
Interesting, I thought this only affected the REPL, but looking at your linked issue in ts2fable seems it's also a problem at least in node apps. I'll have a look, thanks for reporting!

**Comment by inosik** [comment]
Should we revamp the `printf` functions? I looked at them when I wanted to fix #2336, but all the replacing that is done there felt kinda fragile.

I'd either try to port the `Printf` module from FSharp.Core or reuse another package. There are a couple of printf implementations for JS.

**Comment by alfonsogarciacaro** [comment]
Sorry @inosik, missed your comment! 😅 Yes, you're right, the code for print formatting is quite messy. TBH, I'm a bit afraid of touching it because the F# compiler generates a lot of closures and it's a bit difficult to deal with. Not sure if this is done in a "standard way" that would be compatible with some JS package but maybe trying to port the `Printf` module from FSharp.Core would a be a good idea, in line with the current efforts to convert more code in fable-library to F#. If you want to give it you can open a PR and we can follow up there (it would be also nice if we could also convert interpolation into string addition at compile time for simple cases).
